common/src/jni/main/cpp/conscrypt/native_crypto.cc

```diff
@@ -3774,6 +3775,485 @@ static jint NativeCrypto_EVP_AEAD_CTX_open_buf(JNIEnv* env, jclass, jlong evpAea
+    JNI_TRACE("EVP_HPKE_AEAD_aes_128_gcm => ctx=%p", ctx);
+    return reinterpret_cast<jlong>(ctx);
+}
+
+static jlong NativeCrypto_EVP_HPKE_AEAD_aes_256_gcm(JNIEnv* env, jclass) {
+    CHECK_ERROR_QUEUE_ON_RETURN;
+    const EVP_HPKE_AEAD* ctx = EVP_hpke_aes_256_gcm();
+    JNI_TRACE("EVP_HPKE_AEAD_aes_256_gcm => ctx=%p", ctx);
+    return reinterpret_cast<jlong>(ctx);
+}
+
+static jlong NativeCrypto_EVP_HPKE_AEAD_chacha20_poly1305(JNIEnv* env, jclass) {
+    CHECK_ERROR_QUEUE_ON_RETURN;
+    const EVP_HPKE_AEAD* ctx = EVP_hpke_chacha20_poly1305();
+    JNI_TRACE("EVP_HPKE_AEAD_chacha20_poly1305 => ctx=%p", ctx);
+    return reinterpret_cast<jlong>(ctx);
+}
+
+static jbyteArray NativeCrypto_EVP_HPKE_CTX_export(JNIEnv* env, jclass, jobject hpkeCtxRef,
+                                                   jbyteArray exporterCtxArray, jint length) {
+    CHECK_ERROR_QUEUE_ON_RETURN;
+    EVP_HPKE_CTX* hpkeCtx = fromContextObject<EVP_HPKE_CTX>(env, hpkeCtxRef);
+    JNI_TRACE("EVP_HPKE_CTX_export(%p, %p, %d)", hpkeCtx, exporterCtxArray, length);
+
+    if (hpkeCtx == nullptr) {
+        // NullPointerException thrown while calling fromContextObject
+        return {};
+    }
+
+    jbyte* exporterCtxArrayElement = nullptr;
+    const uint8_t* exporterCtx = nullptr;
+    size_t exporterCtxLen = 0;
+    if (exporterCtxArray != nullptr) {
+        exporterCtxArrayElement = env->GetByteArrayElements(exporterCtxArray, 0);
+        exporterCtx = reinterpret_cast<const uint8_t*>(exporterCtxArrayElement);
+        exporterCtxLen = env->GetArrayLength(exporterCtxArray);
+    }
+
+    size_t exportedLen = length;
+    std::vector<uint8_t> exported(exportedLen);
+
+    if (!EVP_HPKE_CTX_export(/* ctx= */ hpkeCtx,
+                             /* out= */ exported.data(),
+                             /* secret_len= */ exportedLen,
+                             /* context= */ exporterCtx,
+                             /* context_len= */ exporterCtxLen)) {
+        if (exporterCtxArrayElement != nullptr) {
+            env->ReleaseByteArrayElements(exporterCtxArray, exporterCtxArrayElement, JNI_ABORT);
+        }
+        conscrypt::jniutil::throwExceptionFromBoringSSLError(env, "EVP_HPKE_CTX_export");
+        return {};
+    }
+
+    if (exporterCtxArrayElement != nullptr) {
+        env->ReleaseByteArrayElements(exporterCtxArray, exporterCtxArrayElement, JNI_ABORT);
+    }
+
+    jbyteArray exportedArray = env->NewByteArray(exportedLen);
+    env->SetByteArrayRegion(exportedArray, 0, exportedLen,
+                            reinterpret_cast<const jbyte*>(exported.data()));
+    return exportedArray;
+}
+
+static void NativeCrypto_EVP_HPKE_CTX_free(JNIEnv* env, jclass, jlong hpkeCtxRef) {
+    CHECK_ERROR_QUEUE_ON_RETURN;
+    EVP_HPKE_CTX* ctx = reinterpret_cast<EVP_HPKE_CTX*>(hpkeCtxRef);
+    JNI_TRACE("EVP_HPKE_CTX_free(%p)", ctx);
+    if (ctx == nullptr) {
+        conscrypt::jniutil::throwNullPointerException(env, "ctx == null");
+        return;
+    }
```

Comment:
Optional: All BoringSSL's free functions already handle NULL (by silently doing nothing, like C `free`), so you can safely remove this. But if you specifically want to throw NPE if someone passes null from Java, checking works.